optimize/client/src/modules/components/ReportRenderer/NoDataNotice.tsx

```diff
@@ -8,35 +8,42 @@
 
 import {ReactNode} from 'react';
 import classnames from 'classnames';
+import {Information, Misuse, Warning} from '@carbon/icons-react';
 
-import {Icon} from 'components';
 import {t} from 'translation';
 
 import './NoDataNotice.scss';
 
 interface NoDataNoticeProps {
-  type?: string;
+  type?: 'error' | 'warning' | 'info' | (string & {}); // this gives us suggestions but also allows any string
```

Comment:
Why do we need the string case?

Reply:
before it was just expecting a string so I though it was done for a purpose, but if you are asking me this question then I guess just the options without allowing any other string is OK